df_py/volume/calc_rewards.py

```diff
@@ -1,505 +1,92 @@
-    stakes: Dict[int, Dict[str, Dict[str, float]]],
-    nftvols_USD: Dict[int, Dict[str, str]],
-) -> List[Tuple[int, str]]:
-    """
-    @arguments
-      stakes - dict of [chainID][nft_addr][LP_addr] : veOCEAN_float
-      nftvols_USD -- dict of [chainID][nft_addr] : vol_USD_float
-
-    @return
-      chain_nft_tups -- list of (chainID, nft_addr), indexed by j
-    """
-    chainIDs = list(stakes.keys())
-    nft_addrs = _get_nft_addrs(nftvols_USD)
-    chain_nft_tups = [
-        (chainID, nft_addr)  # all (chain, nft) tups with stake
-        for chainID in chainIDs
-        for nft_addr in nft_addrs
-        if nft_addr in stakes[chainID]
-    ]
-    return chain_nft_tups
-
-
-@enforce_types
-def _get_nft_addrs(nftvols_USD: Dict[int, Dict[str, str]]) -> List[str]:
-    """
-    @arguments
-      nftvols_USD -- dict of [chainID][nft_addr] : vol_USD_float
-
-    @return
-      nft_addrs -- list of unique nft addrs. Order is consistent.
-    """
-    nft_addrs = set()
-    for chainID in nftvols_USD:
-        for nft_addr in nftvols_USD[chainID]:
-            nft_addrs.add(nft_addr)
-    return sorted(nft_addrs)
-
-
-@enforce_types
-def _get_lp_addrs(stakes: Dict[int, Dict[str, Dict[str, float]]]) -> List[str]:
-    """
-    @arguments
-      stakes - dict of [chainID][nft_addr][LP_addr] : veOCEAN_float
-
-    @return
-      LP_addrs -- list of unique LP addrs. Order is consistent.
-    """
-    LP_addrs = set()
-    for chainID in stakes:
-        for nft_addr in stakes[chainID]:
-            for LP_addr in stakes[chainID][nft_addr]:
-                LP_addrs.add(LP_addr)
-    return sorted(LP_addrs)
-
-
-@enforce_types
-def flatten_rewards(rewards: Dict[int, Dict[str, float]]) -> Dict[str, float]:
-    """
-    @arguments
-      rewards -- dict of [chainID][LP_addr] : reward_float
-
-    @return
-      flat_rewards -- dict of [LP_addr] : reward_float
-    """
-    flat_rewards = {}
-    for chainID in rewards:
-        for LP_addr in rewards[chainID]:
-            if LP_addr not in flat_rewards:
-                flat_rewards[LP_addr] = 0.0
-            flat_rewards[LP_addr] += rewards[chainID][LP_addr]
-    return flat_rewards
-
-
-def merge_rewards(*reward_dicts):
-    merged_dict = {}
+        prev_week = get_df_week_number(start_date)
+
+    if tot_ocean is None:
+        tot_ocean = 0.0
+
+    if do_pubrewards is None:
+        do_pubrewards = DO_PUBREWARDS
+
+    if do_rank is None:
+        do_rank = DO_RANK
+
+    rewperlp, rewinfo = calc_volume_rewards(
+        S,
+        V,
+        C,
+        SYM,
+        R,
+        prev_week,
+        tot_ocean,
+        do_pubrewards,
+        do_rank,
+    )
 
-    for reward_dict in reward_dicts:
-        for key, value in reward_dict.items():
-            merged_dict[key] = merged_dict.get(key, 0) + value
+    csvs.save_volume_rewards_csv(rewperlp, str(csv_dir))
+    csvs.save_volume_rewardsinfo_csv(rewinfo, str(csv_dir))
 
-    return merged_dict
+    return rewperlp, rewinfo
```

Comment:
This should not return anything in memory, because its "return" value is the csvs. Please remove.

Csv in, csv out, that's all.